### PR TITLE
Bookmarks update

### PR DESCRIPTION
**Pull request by rojkov**
Also this PR makes the bookmarks ordered.

### CODE REVIEW

File: src/declarativebookmarkmodel.cpp [hunk 1]
```diff
@@ -47,12 +51,12 @@ void DeclarativeBookmarkModel::removeBookmark(const QString& url)
         return;
     }
 
-    if (bookmarks.contains(url)) {
-        int index = bookmarkUrls.indexOf(url);
+    if (bookmarkIndexes.keys().contains(url)) {
```
Comment (rainemak) [inline src/declarativebookmarkmodel.cpp]: What if we'd drop also bookmarkIndexes QMap. And remove the first occurrence by first checking indexOf(url) if greater than zero then remove it.

We're always removing first occurrence.

Comment (rainemak) [inline src/declarativebookmarkmodel.cpp]: What if we'd drop also bookmarkIndexes QMap. And remove the first occurrence by first checking bookmarkIndex(url) if greater than zero then remove it. We'd need a simple function bookmarkIndex function to find the occurrence.

We're always removing first occurrence.

Comment (rainemak) [inline src/declarativebookmarkmodel.cpp]: And above bookmarkIndex function could be used in DeclarativeBookmarkModel::contains(const QString& url). However, we'd get a performance hit when comparing to map.

Comment (rojkov) [inline src/declarativebookmarkmodel.cpp]: Basically the performace is the only reason why I decided to stick to QMap: O(N) vs O(log2 N).
Do we want to sacrifice that?

File: content/bookmarks.json [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
+        "title": "Jolla",
+        "url": "http://jolla.com/"
     },
     {
         "favicon": "image://theme/icon-m-service-jolla",
-        "title": "How-to tutorial videos",
-        "url": "http://m.youtube.com/playlist?list=PLQgR2jhO_J0y8YSSvVd-Mg9LM88W0aIpD&desktop_uri=%2Fplaylist%3Flist%3DPLQgR2jh
-O_J0y8YSSvVd-Mg9LM88W0aIpD"
+        "title": "How-to videos",
+        "url": "http://m.youtube.com/playlist?list=PLQgR2jhO_J0y8YSSvVd-Mg9LM88W0aIpD&desktop_uri=%2Fplaylist%3Flist%3DPLQgR2jhO_J0y8YSSvVd-Mg9LM88W0aIpD"
```
Comment (rainemak) [inline content/bookmarks.json]: Should be:
https://www.youtube.com/playlist?list=PLQgR2jhO_J0zrl6kLWusekAFr9aHR9jcH

**Comment by rainemak** [comment]
Add some small comments. Other than that LGTM.

Let's keep bookmark indexes as a separate entity.

**Comment by rojkov** [comment]
outdated URL is a good catch! I somehow missed it.
Added comments.

**Pull request merged**